src/NHSD.BuyingCatalogue.Organisations.Api/Services/CreateOrganisationRequest.cs

```diff
@@ -0,0 +1,26 @@
+﻿using NHSD.BuyingCatalogue.Organisations.Api.Models;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Services
+{
+    public sealed class CreateOrganisationRequest
+    {
+        public string OrganisationName { get; }
+
+        public string OdsCode { get; }
+
+        public string PrimaryRoleId { get; }
+
+        public bool CatalogueAgreementSigned { get; }
+
+        public Address Address { get; }
+
+        public CreateOrganisationRequest(string organisationName, string odsCode, string primaryRoleId, bool catalogueAgreementSigned, Address address)
```

Comment:
There's a few too many constructor parameters here for my liking but I understand why you've taken this approach. However, I'm not sure the class really needs to be immutable...

Reply:
I like it being immutable. On the other hand, I get your point, it might be a bit of a pain only being to instantiate this class using the constructor, however, given the purpose of this class, I think immutability suits it well.